src/Commands/Plugin/PluginBaseCommand.php

```diff
@@ -0,0 +1,334 @@
+                    $home = getenv('HOMEPATH');
+                }
+                $home = str_replace('\\', $slash, $home);
+                $plugins_dir = $home . $slash . 'terminus' . $slash . 'plugins' . $slash;
+            } else {
+                $plugins_dir = $home . '/.terminus/plugins/';
+            }
+        } else {
+            // Make sure the proper trailing slash(es) exist
+            $chars = $windows ? 2 : 1;
+            if (substr("$plugins_dir", -$chars) != $slash) {
+                $plugins_dir .= $slash;
+            }
+        }
+        // Create the directory if it doesn't already exist
+        if (!is_dir("$plugins_dir")) {
+            mkdir("$plugins_dir", 0755, true);
+        }
+        return $plugins_dir . $plugin;
+    }
+
+    /**
+     * Get the plugin installation method.
+     *
+     * @param string Plugin name
+     * @return string Plugin installation method
+     */
+    protected function getInstallMethod($plugin)
+    {
+        $slash = $this->getSlash();
+        $plugin_dir = $this->getPluginDir($plugin);
+        $git_dir = $plugin_dir . $slash . '.git';
+        if (is_dir("$git_dir") && $this->commandExists('git')) {
+            return 'git';
+        }
+        $composer_json = $plugin_dir . $slash . 'composer.json';
+        if (file_exists($composer_json)) {
+            return $this->commandExists('composer') ? 'composer' : 'archive';
+        }
+        return 'unknown';
+    }
+
+    /**
+     * Update a specific plugin.
+     *
+     * @param string $plugin Plugin name
+     */
+    protected function updatePlugin($plugin)
+    {
+        $plugins_dir = $this->getPluginDir();
+        $plugin_dir = $plugins_dir . $plugin;
+        if (!is_dir("$plugin_dir")) {
+            $message = "{$plugin} is not installed.";
+            throw new TerminusNotFoundException($message);
+        }
+        $messages = array();
+        $message = "Updating {$plugin}...";
+        $this->log()->notice($message);
+        $method = $this->getInstallMethod($plugin);
+        switch ($method) {
+            case 'git':
+                $installed_version = $this->getInstalledVersion($plugin_dir);
+                $latest_version = $this->getLatestVersion($plugin_dir);
+                if ($installed_version < $latest_version) {
+                    exec("cd \"$plugin_dir\" && git checkout {$latest_version}", $messages);
+                } else {
+                    $messages[] = "Already up-to-date.";
+                }
+                break;
+
+            case 'composer':
+                $composer_info = $this->getComposerInfo($plugin);
+                $project = $composer_info['name'];
+                $packagist_url = "https://packagist.org/packages/{$project}";
+                if ($this->isValidUrl($packagist_url)) {
+                    exec("rm -rf \"{$plugin_dir}\" && composer create-project --prefer-source --keep-vcs -n -d {$plugins_dir} {$project}:~1", $messages);
```

Comment:
Oh, okay, got it. So if you use the install command to add the plugin, and it runs `composer create-project...`, then the update command will follow the `git` branch here. That makes sense. It might be beneficial to continue to support multiple update methods here, even if only one installation method is supported, just to make things easy for folks who may have installed without the tool.